Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1235,6 +1250,17 @@ PhysicalParticleContainer::PushPX (WarpXParIter& pti,
     // Add guard cells to the box.
     box.grow(ngEB);
 
+    // Auxiliary booleans
+    bool const gather_fields = (
+        !do_not_gather &&
+        (position_push_type == DtType::Full || position_push_type == DtType::FirstHalf)
```

Comment:
Maybe replace this line with `momentum_push_type != DtType::None`?

Reply:
Thanks! Yes, I think it should work, and it's more clear:
```suggestion
        momentum_push_type != DtType::None
```